tsrc/executor.py

```diff
@@ -3,12 +3,17 @@
 """
```

Comment:
This module needs a big comment to explain why and how we use Outcome, Task, OutcomeCollection and Executors

There's also a bunch of missing docstrings.